Submit shields report with origin + path

deeppandya opened the pull request:
<!-- Add brave-browser issue below that this PR will resolve -->
Resolves https://github.com/brave/brave-browser/issues/32640

<!-- CI-related labels that can be applied to this PR:
* CI/run-audit-deps (1) - check for known npm/cargo vulnerabilities (audit_deps)
* CI/run-network-audit (1) - run network-audit
* CI/run-upstream-tests - run Chromium unit and browser tests on Linux and Windows (otherwise only on Linux)
* CI/skip - do not run CI builds (except noplatform)
* CI/skip-linux-arm64, CI/skip-linux-x64, CI/skip-android, CI/skip-macos, CI/skip-ios, CI/skip-windows-arm64, CI/skip-windows-x64, CI/skip-windows-x86 - skip CI builds for specific platforms
* CI/skip-upstream-tests - do not run Chromium unit, or browser tests (otherwise only on Linux)
* CI/skip-all-linters - do not run presubmit and lint checks
* CI/storybook-url (1) - deploy storybook and provide a unique URL for each build

(1) applied automatically when some files are changed (see: https://github.com/brave/brave-core/blob/master/.github/labeler.yml)
-->

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

deeppandya commented on android/java/org/chromium/chrome/browser/shields/BraveShieldsHandler.java:
```diff
@@ -821,7 +821,7 @@ public void onClick(View view) {
                         NTPBackgroundImagesBridge.getInstance(mProfile).getReferralApiKey();
                 BraveShieldsUtils.BraveShieldsWorkerTask mWorkerTask =
-                        new BraveShieldsUtils.BraveShieldsWorkerTask(mTitle, referralApiKey);
+                        new BraveShieldsUtils.BraveShieldsWorkerTask(mHost, referralApiKey);
```
> Not a point of the PR, but there is some confusing naming at 
> 
> https://github.com/brave/brave-core/blob/master/android/java/org/chromium/chrome/browser/shields/BraveShieldsHandler.java#L277-L278
> 
> ```Java
>         mHost = tab.getUrl().getSpec();
>         mTitle = tab.getUrl().getHost();
> ```
> So maybe the good time now to rename `mHost` => `mUrlSpec`?
> 

That's exactly what I wanted to refactor but we are planning to revamp the whole thing so I didn't want to update it in this branch.

pes10k replied:
the above is confusing to me too.  @deeppandya can you confirm that this results in sending `origin + path` and not the full URL?

If so, looks great to me

deeppandya commented:
@pes10k i have refactored the variables with correct context here : https://github.com/brave/brave-core/pull/19958/commits/ab1edc842efb74756d97a38438976c93f6562175
and updated URL based on https://github.com/brave/brave-core/pull/12039/files#diff-88d92384c89cf019409cb1a5a8339d20638c8b9ef7f89f4f69c056925eba56faR44

deeppandya commented:
@pes10k i am not showing the origin+path on the UI as it would be too big for the mobile UI. but when user tries to submit the report, i use origin+path to submit the report. Should i create a security/privacy review for it ?

pes10k commented:
@deeppandya i think its important to make clear what we're sending to the user.  Is it possible to show the full URL truncated (i.e, `https://example.com/very_very_long_path` becomes `https://example.com/very_very…`) until clicked on?

Otherwise, can can we stick it in a <textarea> with a scroll bar or similar?

There just needs to be someway for the user to see the full URL before agreeing to send it.  Otherwise, we'd need a privacy and security review

deeppandya commented:
@pes10k here is the updated UI : 

https://github.com/brave/brave-core/assets/5146988/845ff3d8-7028-41c0-a5b1-80b3a29553db

Please let me know if you have any concerns.

mkarolin commented on browser/ui/android/strings/android_brave_strings.grd:
```diff
@@ -4047,6 +4044,9 @@ If you don't accept this request, VPN will not reconnect and your internet conne
     <message name="IDS_MENU_BRAVE_LEO" desc="Brave AI menu item text">
       Brave Leo
     </message>
+    <message name="IDS_ELLIP_TEXT" desc="Generic ellip text" translateable="false">
```
Nit: use full word `ellipsis` in both the `name` and the `desc` attributes.

mkarolin reviewed (approved):
`strings`++

kjozwiak commented:
Verification **PASSED** on `Pixel 6` running `Android 14` using the following build(s):

```
Brave | 1.61.49 Chromium: 119.0.6045.33 (Official Build) canary (64-bit)
--- | ---
Revision | 0a5ed9b4662abaa91373eb8d17f5b616013ca743
OS | Android 14; Build/U1B2.230922.010; 34; REL
```

Using the STR/information outlined via https://github.com/brave/brave-browser/issues/32640#issue-1874917038, went through the following:

* logged into FB and loaded a group (one that I've personally joined)
* disabled the the shields via the panel on `m.facebook.com`
* once the shields were disabled, clicked on `Report a broken site`
* ensured that the `Report a broken site` displays the correct link that's being reported
* clicked `Submit` and ensured that the request correctly went through `https://webcompat.brave.com/1/webcompat`

**`1.59.120 Chromium: 118.0.5993.88`** | **`1.61.49 Chromium: 119.0.6045.33`**
---------------------------------------------|--------------------------------------------
![Screenshot_20231024-214350](https://github.com/brave/brave-core/assets/2602313/59b9e082-132f-4d99-8ffa-1a30f43e44ca) | ![Screenshot_20231024-213206](https://github.com/brave/brave-core/assets/2602313/f72e8457-d161-49ab-97d8-8c76794ec71c)

#### `Quick Note:` Received a `500 internal error` whenever submitting but I'm pretty sure that's related to Cloudfront once SSL Proxy is enabled via `Proxyman`. Once you disable `SSL Proxy`, the request goes through fine but can't be inspected due to being `HTTPS`. However, you can clearly see that we're sending `origin + path` as per the `domain` in the screenshot.

**`1.59.120 Chromium: 118.0.5993.88`** | **`1.61.49 Chromium: 119.0.6045.33`**
---------------------------------------------|--------------------------------------------
![Screenshot 2023-10-24 at 9 44 33 PM](https://github.com/brave/brave-core/assets/2602313/fb5f8209-0950-4bc0-aa54-14a0545d9aa3) | ![Screenshot 2023-10-24 at 9 45 26 PM](https://github.com/brave/brave-core/assets/2602313/3b7a0147-f6e0-4963-901e-ae3c82cd7ad9)
